input/kafka.go

```diff
@@ -0,0 +1,114 @@
+package input
+
+import (
+	"context"
+	"github.com/Shopify/sarama"
+	log "github.com/sirupsen/logrus"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+	"github.com/graphite-ng/carbon-relay-ng/cfg"
+)
+
+
+
+type Kafka struct {
+	config cfg.Config
+	dispatcher Dispatcher
+	client sarama.ConsumerGroup
+	consumer Consumer
+	ctx context.Context
+}
+
+func (kafka *Kafka) Name() string {
+	return "kafka"
+}
+
+
+func (kafka *Kafka) Start() error {
+	go func() {
+		for {
+			kafka.consumer.ready = make(chan bool, 0)
+			err := kafka.client.Consume(kafka.ctx, strings.Fields(kafka.config.Kafka.Kafka_topic), &kafka.consumer)
+			if err != nil {
+				panic(err)
+			}
+		}
+	}()
+	<-kafka.consumer.ready // Await till the consumer has been set up
+	log.Infoln("Sarama consumer up and running!...")
+
+	sigterm := make(chan os.Signal, 1)
+	signal.Notify(sigterm, syscall.SIGINT, syscall.SIGTERM)
+
+	<-sigterm // Await a sigterm signal before safely closing the consumer
+
+	err := kafka.client.Close()
+
+	return err
+}
+
+func (kafka *Kafka) Stop() bool {
+	err := kafka.client.Close()
```

Comment:
Add a shutdown chan, that you close at this line

And check it in the for_range loop to avoid consuming after Stop()